plugins/module_utils/network/sonic/config/ntp/ntp.py

```diff
@@ -0,0 +1,550 @@
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import re
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.cfg.base import (
+    ConfigBase,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    to_list,
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.facts.facts import Facts
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic import (
+    to_request,
+    edit_config
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.utils import (
+    get_diff,
+    update_states,
+    normalize_interface_name,
+    normalize_interface_name_list
+)
+from ansible.module_utils.connection import ConnectionError
+
+PATCH = 'PATCH'
+DELETE = 'DELETE'
+
+TEST_KEYS_ADD = [
+    {
+        "vrf": "", "enable_ntp_auth": "",
+        "source_interfaces": "", "trusted_keys": "",
+        "servers": {"address": "", "key_id": "", "maxpoll": "", "minpoll": ""},
+        "ntp_keys": {"key_id": "", "key_type": "", "key_value": "", "encrypted": ""}
+    }
+]
+
+TEST_KEYS_DEL = [
+    {
+        "vrf": "", "enable_ntp_auth": "", "source_interfaces": "", "trusted_keys": "",
+        "servers": {"address": ""}, "ntp_keys": {"key_id": ""}
+    }
+]
+
+
+class Ntp(ConfigBase):
+    """
+    The sonic_ntp class
+    """
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'ntp',
+    ]
+
+    def __init__(self, module):
+        super(Ntp, self).__init__(module)
+
+    def get_ntp_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        ntp_facts = facts['ansible_network_resources'].get('ntp')
+
+        if not ntp_facts:
+            return []
+        return ntp_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+        commands = list()
+        requests = list()
+
+        existing_ntp_facts = self.get_ntp_facts()
+
+        commands, requests = self.set_config(existing_ntp_facts)
+
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    if not re.search("code.*404", str(exc)):
```

Comment:
Why is "resource not found" excluded from providing a failure message?

Reply:
This is for DELETE case. It may not be necessary.

Reply:
It shouldn't be necessary. If the configuration to be deleted isn't present, no "request" should be generated for that item. If a 404 error occurs, it indicates a code bug that needs to be investigated.

Reply:
Removed 404 error exclusion.

Reply:
Verified.